### PR TITLE
Pushbutton tweaks

### PR DESCRIPTION
**Pull request by ronso0**
I'd like to improve our custom pushbuttons a bit because the limitations are a bit annoying, for example with pushbuttons in the skin settings menu.

**Also restyle text labels when button is hovered by cursor**
Previously, `#MyPushbutton:hover` styles wouldn't be applied to labels. To do this with `#MyPushbutton[hover="true"]` I added a custom `hover` property which is set by overriding QWidget's enterEvent() and leaveEvent() functions.

**Elide long labels**
Now, when parsing the pushbutton template the `<Elide>` node is used to apply the elide mode to long labels. To make this fully usable, I also want the button to consider the applied `padding` from external stylesheets and adjust the label rectangle. This helps to keep the labels within the visible boundary of backpaths or border images.

The actual goal with the elide mode is being able to **use effect parameter names as button label**.
This allows to have compact parameter buttons without additional labels. (in Tango I used button/label stacks for that but they're buggy)
I didn't investigate that, yet, and I guess I need a little help with it in order to mak it efficient without too much code duplication from `WEffectButtonParameter`.
My idea is to add a `setText(QString)` slot to wpushbutton.cpp which can be used by weffectpushbutton.cpp if legacyskinparser finds `<ParameterNameLabel>true</ParameterNameLabel>`, for example.

- [x] add `hover` property
- [x] read `<Elide>` mode
- [x] ~~consider `padding` from stylesheet~~ dropped. too complicated for too little effect
- [x] ~~effect parameter buttons with "ParameterName" labels~~ postponed
- [ ] remove sample skin

### CODE REVIEW

File: src/widget/wpushbutton.h [hunk 1]
```diff
@@ -97,9 +108,12 @@ class WPushButton : public WWidget {
 
     // True, if the button is currently pressed
     bool m_bPressed;
+    // True, if the button is pointer is above button
+    bool m_bHovered;
 
     // Array of associated pixmaps
     int m_iNoStates;
+    Qt::TextElideMode m_elideMode;
```
Comment (daschuer) [inline src/widget/wpushbutton.h]: not initialized in the constructor.

File: src/widget/wpushbutton.cpp [hunk 2]
```diff
@@ -337,23 +355,43 @@ void WPushButton::paintEvent(QPaintEvent* e) {
...
 
     if (m_leftButtonMode == ControlPushButton::POWERWINDOW
             && m_iNoStates == 2) {
         if (leftClick) {
-            m_clickTimer.setSingleShot(true);
-            m_clickTimer.start(ControlPushButtonBehavior::kPowerWindowTimeMillis);
-
-            double emitValue = getControlParameterLeft() == 0.0 ? 1.0 : 0.0;
-            setControlParameterLeftDown(emitValue);
+            if (getControlParameterLeft() == 0.0) {
```
Comment (daschuer) [inline src/widget/wpushbutton.cpp]: Is this related to this PR? What does it solve?

**Review by daschuer — CHANGES_REQUESTED** [review]
Thank you. I have left some comments.

File: src/widget/wpushbutton.cpp (hunk 2 above)
Comment (ronso0) [reply src/widget/wpushbutton.cpp]: oh, not intended. this must have slipped in from an unrelated branch.
I'll revert it.

File: src/widget/wpushbutton.h (hunk 1 above)
Comment (ronso0) [reply src/widget/wpushbutton.h]: is it okay to do this in [setState()](https://github.com/mixxxdj/mixxx/pull/2471/files?file-filters%5B%5D=.cpp&file-filters%5B%5D=.h#diff-78d5371c502303528e220b3605c53534R251)?

Comment (daschuer) [reply src/widget/wpushbutton.h]: No, I guess code rule checkers will complain anyway. 
Add it just here: 
https://github.com/mixxxdj/mixxx/blob/be526c6ccfc30d78dfdd0cb65c80dffedb9e34e7/src/widget/wpushbutton.cpp#L39

Comment (ronso0) [reply src/widget/wpushbutton.h]: ..and here https://github.com/mixxxdj/mixxx/blob/be526c6ccfc30d78dfdd0cb65c80dffedb9e34e7/src/widget/wpushbutton.cpp#L47
the other variables are initialized in setStates(). To reduce code duplication?

Comment (daschuer) [reply src/widget/wpushbutton.h]: Ups, yes setStates() is a good place for this. I was not aware that it is called from every constructor.

**Review by daschuer — APPROVED** [review]

**Comment by daschuer** [comment]
Thanks. Please give a hint when this is merge able.

**Comment by ronso0** [comment]
I just screwed up my local git folder by a veeery stupid action. Unfortunately I won't find the time to fix this (setup remotes, branches, 2FA, ssh and stuff) until end of february, 3 weeks from now, because I'm gonna be travelling.

I postponed the effect name pushbuttons I had in mind, this will take too long.

I'll finish some bits via the website, like removing the sample styles, and let you know when I consider this ready for merge.

**Comment by ronso0** [comment]
Yeah, removing the sample style is the only thing left.

**Comment by ronso0** [comment]
ready.

**Review by daschuer — APPROVED** [review]

**Comment by daschuer** [comment]
Thank you for this nice bit. LGTM.

**Comment by ronso0** [comment]
Seems either this or #2460 broke the debug tooltip for wpushbutton.
I'll have a look..
